### PR TITLE
ValueError: operands could not be broadcast together with shapes #425

### LINKED ISSUE #425: ValueError: operands could not be broadcast together with shapes
```
C:\Users\VargB\AppData\Local\Programs\Python\Python36\lib\site-packages\scipy\stats\stats.py:3005: RuntimeWarning: Mean of empty slice.
  mx = x.mean()
C:\Users\VargB\AppData\Local\Programs\Python\Python36\lib\site-packages\numpy\core\_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
2019-02-20 11:00:53,158 [Analytics] [ERROR]  handle_analytic_task Exception: 'Traceback (most recent call last):
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\analytic_unit_manager.py", line 94, in handle_analytic_task
    result_payload = await self.__handle_analytic_task(task)
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\analytic_unit_manager.py", line 82, in __handle_analytic_task
    return await worker.do_train(payload['segments'], data, payload['cache'])
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\analytic_unit_worker.py", line 28, in do_train
    new_cache: ModelCache = await self._training_feature
  File "C:\Users\VargB\AppData\Local\Programs\Python\Python36\lib\concurrent\futures\thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\detectors\pattern_detector.py", line 46, in train
    new_cache = self.model.fit(dataframe, segments, cache)
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\models\model.py", line 86, in fit
    self.do_fit(dataframe, labeled, deleted, learning_info)
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\models\general_model.py", line 49, in do_fit
    correlation_list = utils.get_correlation(self.state['pattern_center'], self.state['pattern_model'], data, self.state['WINDOW_SIZE'])
  File "C:\Users\VargB\git\hastic-server\analytics\bin\..\analytics\utils\common.py", line 230, in get_correlation
    correlation = pearsonr(labeled_segment, av_model)
  File "C:\Users\VargB\AppData\Local\Programs\Python\Python36\lib\site-packages\scipy\stats\stats.py", line 3008, in pearsonr
    r_num = np.add.reduce(xm * ym)
ValueError: operands could not be broadcast together with shapes (0,) (9,)
```

Steps to reproduce:
Label  2 segments far from each other
image
learning
in analytics we have data with 3285 values
image
segments indexes - 3139, 148
then label some pattern in right part of data
image
in analytics we have data with 2465 values
image
and value error
image

### PR DESCRIPTION
**Pull request by VargBurz**
Closes #425

### CODE REVIEW

File: analytics/analytics/utils/common.py [hunk 1]
```diff
@@ -186,6 +187,9 @@ def find_extremum_index(segment: np.ndarray, selector: bool) -> int:
         return segment.argmin()
 
 def get_interval(data: pd.Series, center: int, window_size: int) -> pd.Series:
+    if center >= len(data):
```
Comment (rozetko) [inline analytics/analytics/utils/common.py]: what if `center + window_size > len(data)`

Comment (amper43) [reply analytics/analytics/utils/common.py]: please, check that `utils.timestamp_to_index` returns correct index

File: analytics/analytics/utils/common.py [hunk 2]
```diff
@@ -227,6 +231,8 @@ def get_correlation(segments: list, av_model: list, data: pd.Series, window_size
         labeled_segment = utils.get_interval(data, segment, window_size)
         labeled_segment = utils.subtract_min_without_nan(labeled_segment)
         labeled_segment = utils.check_nan_values(labeled_segment)
+        if len(labeled_segment) == 0 or len(labeled_segment) != len(av_model):
+            continue
```
Comment (amper43) [inline analytics/analytics/utils/common.py]: please, check case where `correlation_list` still empty after loop

File: analytics/analytics/utils/common.py (hunk 1 above)
Comment (VargBurz) [reply analytics/analytics/utils/common.py]: @rozetko  it is normal situation, analytics can operate with half-patterns

Comment (VargBurz) [reply analytics/analytics/utils/common.py]: @amper43  done

File: analytics/analytics/utils/common.py (hunk 2 above)
Comment (VargBurz) [reply analytics/analytics/utils/common.py]: done

**Pull request merged**